rdr_service/dao/bq_questionnaire_dao.py

```diff
@@ -25,6 +29,59 @@ def make_bqrecord(self, p_id, module_id, latest=False, convert_to_enum=False):
         :param convert_to_enum: If schema field description includes Enum class info, convert value to Enum.
         :return: BQTable object, List of BQRecord objects
         """
+
+        # Query to get all the question codes across all questionnaire versions of the module.
+        _question_code_sql = """
+            -- select c.code_id, c.value
+            select c.value
+            from code c
+            inner join (
+                select distinct qq.code_id
+                from questionnaire_question qq where qq.questionnaire_id in (
+                    select qc.questionnaire_id from questionnaire_concept qc
+                            where qc.code_id = (
+                                select code_id from code c2 where c2.value = :module_id )
+                )
+            ) qq2 on qq2.code_id = c.code_id
+            order by c.code_id;
+        """
+
+        # Query to get a list of questionnaire_response_id values for this participant and module
+        _participant_module_responses_sql = """
+            select qr.questionnaire_id, qr.questionnaire_response_id, qr.created, qr.authored, qr.language,
+                   qr.participant_id
+            from questionnaire_response qr
+            where qr.participant_id = :p_id and questionnaire_id IN (
+                select q.questionnaire_id from questionnaire q
+                inner join questionnaire_history qh on q.version = qh.version
+                       and qh.questionnaire_id = q.questionnaire_id
+                inner join questionnaire_concept qc on qc.questionnaire_id = q.questionnaire_id
+                inner join code c on c.code_id = qc.code_id
+                where c.value = :module_id
+                order by qr.created DESC
+            );
+        """
+
+        # Query to get questionnaire answers for a specific questionnaire_response_id
+        _response_answers_sql = """
+            SELECT qr.questionnaire_id,
+               qq.code_id,
+               (select c.value from code c where c.code_id = qq.code_id) as code_name,
+               COALESCE((SELECT c.value from code c where c.code_id = qra.value_code_id),
+                        qra.value_integer, qra.value_decimal,
+                        qra.value_boolean, qra.value_string, qra.value_system,
+                        qra.value_uri, qra.value_date, qra.value_datetime) as answer
+            FROM questionnaire_response qr
+                     INNER JOIN questionnaire_response_answer qra
+                                ON qra.questionnaire_response_id = qr.questionnaire_response_id
+                     INNER JOIN questionnaire_question qq
+                                ON qra.question_id = qq.questionnaire_question_id
+                     INNER JOIN questionnaire q
+                                ON qq.questionnaire_id = q.questionnaire_id
```

Comment:
Is the questionnaire being joined in for anything? I was trying to trace back to what joining it this way might affect, but it doesn't seem like any data is read from the `questionnaire` table.

Reply:
These were the queries we worked up in the slack meeting, cut and pasted.    I'll review them again

Reply:
Fixed in most recent changes